lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -206,6 +207,46 @@ end
 @inline batch_size(::BatchMixedDuplicated{T,N}) where {T,N} = N
 @inline batch_size(::Type{BatchMixedDuplicated{T,N}}) where {T,N} = N
 
+"""
+    batchify_activity(::Type{A}, ::Val{B})
+
+Turn an activity (or [`Annotation`](@ref)) type `A` into the correct activity type for a batch of size `B`.
+
+# Examples
+
+```jldoctest
+julia> using EnzymeCore
+
+julia> EnzymeCore.batchify_activity(Active{Float64}, Val(2))
+Active{Float64}
+
+julia> EnzymeCore.batchify_activity(Duplicated{Vector{Float64}}, Val(2))
+BatchDuplicated{Vector{Float64}, 2}
+```
+"""
+batchify_activity(::Type{Active{T}}, ::Val{B}) where {T,B} = Active{T}
```

Comment:
not that it matters much here, but I feel like we do have a missing BatchActive -- in particular this is already the case for certain custom rules of batched ?returns (I don't remmeber specifics) that can't currently be expressed at the moment

Reply:
Could it be the cause for #2514?